cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -439,6 +472,255 @@ struct ScalarMinMax {
+          }
+        } else {
+          const auto& array = *batch[col].array();
+          if (!array.MayHaveNulls() ||
+              bit_util::GetBit(array.buffers[0]->data(), array.offset + row)) {
+            const auto offsets = array.GetValues<offset_type>(1);
+            const auto data = array.GetValues<uint8_t>(2, /*absolute_offset=*/0);
+            const int64_t length = offsets[row + 1] - offsets[row];
+            visit_value(
+                string_view(reinterpret_cast<const char*>(data + offsets[row]), length));
+          } else if (!options.skip_nulls) {
+            result = util::nullopt;
+            break;
+          }
+        }
+      }
+
+      if (result) {
+        builder.UnsafeAppend(*result);
+      } else {
+        builder.UnsafeAppendNull();
+      }
+    }
+
+    std::shared_ptr<Array> string_array;
+    RETURN_NOT_OK(builder.Finish(&string_array));
+    *out = *string_array->data();
+    out->mutable_array()->type = batch[0].type();
+    DCHECK_EQ(batch.length, out->array()->length);
+    return Status::OK();
+  }
+
+  // Compute an upper bound for the length of the output batch.
+  static int64_t CalculateOutputSizeUpperBound(const ExecBatch& batch) {
+    int64_t estimated_final_size = 0;
+    for (size_t col = 0; col < batch.values.size(); col++) {
+      const auto& datum = batch[col];
+      if (datum.is_scalar()) {
+        const auto& scalar = checked_cast<const BaseBinaryScalar&>(*datum.scalar());
+        if (scalar.is_valid) {
+          estimated_final_size = std::max(estimated_final_size, scalar.value->size());
+        }
+      } else {
+        DCHECK(datum.is_array());
+        estimated_final_size =
+            std::max(estimated_final_size, datum.array()->buffers[2]->size());
+      }
+    }
+    return estimated_final_size;
+  }
+};
+
+template <typename Op>
+struct FixedSizeBinaryScalarMinMax {
+  static Status Exec(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+    const ElementWiseAggregateOptions& options = MinMaxState::Get(ctx);
+    if (std::all_of(batch.values.begin(), batch.values.end(),
+                    [](const Datum& d) { return d.is_scalar(); })) {
+      return ExecOnlyScalar(ctx, options, batch, out);
+    }
+    return ExecContainingArrays(ctx, options, batch, out);
+  }
+
+  static Status ExecOnlyScalar(KernelContext* ctx,
+                               const ElementWiseAggregateOptions& options,
+                               const ExecBatch& batch, Datum* out) {
+    if (batch.values.empty()) {
+      return Status::OK();
+    }
+    auto output = checked_cast<BaseBinaryScalar*>(out->scalar().get());
+    const size_t num_args = batch.values.size();
+
+    const auto batch_type = batch[0].type();
+    const auto binary_type = checked_cast<const FixedSizeBinaryType*>(batch_type.get());
+    string_view result =
+        UnboxScalar<FixedSizeBinaryType>::Unbox(*batch.values.front().scalar());
+    for (size_t i = 1; i < num_args; i++) {
```

Comment:
I'm curious: why is this not using the same algorithm as for variable-width binary types?

(you could perhaps even share most of the implementation...)